server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -139,25 +157,34 @@ export class TenantManager {
         const db = await this.mongoManager.getDatabase();
         const collection = db.collection<ITenantDocument>(this.collectionName);
 
-        const tenantKey = this.generateTenantKey();
-        const encryptedTenantKey = this.secretManager.encryptSecret(tenantKey);
-        if (encryptedTenantKey == null) {
-            winston.error("Tenant key encryption failed.");
-            Lumberjack.error("Tenant key encryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
-            return Promise.reject(new Error("Tenant key encryption failed."));
+        const tenantKey1 = this.generateTenantKey();
+        const encryptedTenantKey1 = this.secretManager.encryptSecret(tenantKey1);
+        if (encryptedTenantKey1 == null) {
+            winston.error("Tenant key1 encryption failed.");
+            Lumberjack.error("Tenant key1 encryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
+            return Promise.reject(new Error("Tenant key1 encryption failed."));
+        }
+
+        const tenantKey2 = this.generateTenantKey();
+        const encryptedTenantKey2 = this.secretManager.encryptSecret(tenantKey2);
+        if (encryptedTenantKey2 == null) {
+            winston.error("Tenant key2 encryption failed.");
+            Lumberjack.error("Tenant key2 encryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
+            return Promise.reject(new Error("Tenant key2 encryption failed."));
```

Comment:
If the encryption of the secondary key fails, should we reject or should we simply pass secondary key as an empty string, since the primary is fine?

Reply:
We should reject, new created tenants would expect 2 keys. If either of the 2 keys fails, we shouldn't proceed.